Source/WebCore/page/Navigator.cpp

```diff
@@ -247,11 +267,44 @@ void Navigator::initializePluginAndMimeTypeArrays()
 
     auto [publiclyVisiblePlugins, additionalWebVisiblePlugins] = frame->page()->pluginData().publiclyVisiblePluginsAndAdditionalWebVisiblePlugins();
 
+    m_pdfViewerEnabled = frame->loader().client().canShowMIMEType("application/pdf"_s);
+
+    // macOS uses a PDF Plugin (which may be disabled). Other ports handle PDF's through native
+    // platform views outside the engine, or use pdf.js.
+    bool hasEnabledWebKitBuiltinPDFPlugin = frame->page()->pluginData().builtInPDFPlugin().has_value();
+
     Vector<Ref<DOMPlugin>> publiclyVisibleDOMPlugins;
     Vector<Ref<DOMPlugin>> additionalWebVisibleDOMPlugins;
     Vector<Ref<DOMMimeType>> webVisibleDOMMimeTypes;
 
-    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisiblePlugins.size());
+    size_t publiclyVisibleDOMPluginCount = publiclyVisiblePlugins.size();
+#if PLATFORM(COCOA) || ENABLE(PDFJS)
+    if (m_pdfViewerEnabled)
+        publiclyVisibleDOMPluginCount += dummyPDFPluginCount;
+#endif
+
+    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisibleDOMPluginCount);
+
+#if PLATFORM(COCOA) || ENABLE(PDFJS)
+    if (m_pdfViewerEnabled) {
+        auto pluginInfo = hasEnabledWebKitBuiltinPDFPlugin ? *frame->page()->pluginData().builtInPDFPlugin() : PluginData::dummyPDFPluginInfo();
+
+        // https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewing-support
+        // Section 8.9.1.6 states that if pdfViewerEnabled is true, we must return a list
+        // of exactly five PDF view plugins, in a particular order.
+        const auto& builtinName = builtInPDFPluginName();
```

Comment:
builtinName is a super generic name considering we're talking about the built-in PDF plugin specifically.

Reply:
I'll revise that to use an `ASCIILiteral` to represent the 'WebKit built-in PDF' label, and use that in the various places that need to know the string.